secant/Features/App/App.swift

```diff
@@ -43,6 +65,83 @@ extension AppReducer {
+                // TODO: use database URL from the same issue #220
+                let documentsURL = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+                let dataDatabaseURL = documentsURL.appendingPathComponent("ZcashSDK.defaultDataDbName", isDirectory: false)
+                let attributes = try fileManager.attributesOfItem(atPath: dataDatabaseURL.path)
+                let databaseFilesPresent = attributes.isEmpty
+                
+                switch (keysPresent, databaseFilesPresent) {
+                case (false, false):
+                    walletInitialisationState = .uninitialized
+                case (false, true):
+                    walletInitialisationState = .keysMissingFilesPresent
+                case (true, false):
+                    walletInitialisationState = .keysPresentFilesMissing
+                case (true, true):
+                    walletInitialisationState = .initialized
+                }
+            } catch CocoaError.fileNoSuchFile, CocoaError.fileReadNoSuchFile {
+                walletInitialisationState = keysPresent ? .keysPresentFilesMissing : .uninitialized
+            } catch {
+                walletInitialisationState = .failure(ErrorWrapper(error: error))
+            }
+            
+            return Effect(value: .walletInitialisationResult(walletInitialisationState))
+
+            /// All possible responses to the `.checkWalletInitialisation` action result.
+            /// `uninitialized` results in the start of the Onboarding flow for the user.
+        case .walletInitialisationResult(let walletInitialisationState):
+            switch walletInitialisationState {
+            case .uninitialized:
+                return Effect(value: .updateRoute(.onboarding))
+                    .delay(for: 3, scheduler: environment.mainQueue)
+                    .eraseToEffect()
+            case .initialized, .keysPresentFilesMissing:
+                return Effect(value: .initialiseApp)
+            case .keysMissingFilesPresent:
+                state.appInitialisationState = walletInitialisationState
+                // TODO: error we need to handle, issue #221 (https://github.com/zcash/secant-ios-wallet/issues/221)
+                return .none
+            case .failure(error: let error):
+                state.appInitialisationState = walletInitialisationState
+                // TODO: error we need to handle, issue #221 (https://github.com/zcash/secant-ios-wallet/issues/221)
+                return .none
+            }
+
+            /// Stored wallet is present, database files may or may not be present, trying to initialise app state variables and environments.
+            /// When initialisation succeeds user is taken to the home screen.
+        case .initialiseApp:
```

Comment:
In the US I think we standardize on using a `z` in the word initialize.  Can we standardize on that like we did in the other PR?

Reply:
For me is the same but the SDK already uses `z` so looking at it from that perspective, that's a good idea.

Reply:
fixed